src/field.cpp

```diff
@@ -881,6 +881,35 @@ bool map::process_fields_in_submap( submap *const current_submap,
                                 bool destroyed = fuel->burn( frd, can_spread);
 
                                 if( destroyed ) {
+									//Check what the item is made of and spawn appropriate byproducts
+									std::vector<material_id> all_mats = fuel->made_of();
+									if( all_mats.size() > 0 )
+									{
+									    units::mass fuel_weight = fuel->type->weight;
+									    //Items that are multiple materials are assumed to be equal parts each.
+									    units::mass by_weight = fuel_weight / all_mats.size();
+
+									    std::vector<itype_id> all_by;
+									    std::vector<int> all_by_n;
```

Comment:
Why two separate vectors when you apparently need them synchronized (element N in one belongs to element N in the other). Use a single vector containing `std::pair<itype_id, int>` instead.

Actually, why the separate step of gathering type and count and later creating the item? You can create the item directly inside the `if` branch below.

And when you call an `emplace` function on a container, the parameters are forwarded to the constructor of the contained type. You don't need to create the instance yourself. In other words: write `emplace_back( "item_type", turn, charges )` and the container will create an item for you.